app/src/main/java/com/android/sample/ui/fridge/EditFridgeItemScreen.kt

```diff
@@ -0,0 +1,219 @@
+  PlateSwipeScaffold(
+      navigationActions = navigationActions,
+      selectedItem = navigationActions.currentRoute(),
+      content = { paddingValues ->
+        EditComposable(paddingValues, navigationActions, userViewModel)
+      },
+      showBackArrow = true)
+}
+
+/**
+ * Composable function to edit a fridge item
+ *
+ * @param paddingValues The padding values to be applied to the composable.
+ * @param navigationActions The navigation actions to be applied to the composable.
+ * @param userViewModel The view model to be applied to the composable.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("StateFlowValueCalledInComposition")
+@Composable
+fun EditComposable(
+    paddingValues: PaddingValues,
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel
+) {
+
+  val ingredient = userViewModel.ingredientList.collectAsState().value[0].first
+  var quantity by remember { mutableIntStateOf(1) }
+  var expirationDate by remember { mutableStateOf(LocalDate.now()) }
+
+  val showDatePickerDialog = remember { mutableStateOf(false) }
+
+  Column(
+      modifier = Modifier.fillMaxSize().padding(paddingValues).padding(PADDING_16.dp),
+      verticalArrangement = Arrangement.spacedBy(PADDING_16.dp)) {
+        // Ingredient Name
+        Text(
+            text = ingredient.name,
+            modifier = Modifier.padding(PADDING_16.dp),
+            style = MaterialTheme.typography.titleLarge,
+            lineHeight = TITLE_LINE_HEIGHT.sp,
+            fontSize = TITLE_FONT_SIZE.sp,
+            color = MaterialTheme.colorScheme.onPrimary)
+
+        // Ingredient Image
+        Image(
+            painter =
+                rememberAsyncImagePainter(
+                    model = ingredient.images[PRODUCT_FRONT_IMAGE_THUMBNAIL_URL]),
+            contentDescription = stringResource(R.string.recipe_image),
+            alignment = Alignment.Center,
+            modifier =
+                Modifier.fillMaxSize()
+                    .weight(4f)
+                    .testTag(RECIPE_IMAGE_1)
+                    .clip(RoundedCornerShape(INGREDIENT_DISPLAY_IMAGE_BORDER_RADIUS.dp)),
+            contentScale = ContentScale.Fit,
+        )
+
+        // Quantity Adjustment
+        Row(
+            modifier = Modifier.fillMaxSize().weight(2f),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.spacedBy(PADDING_16.dp)) {
+              Text(
+                  text = "Quantity (x ${ingredient.quantity}):",
+                  modifier = Modifier.weight(1f),
+                  style = MaterialTheme.typography.bodyMedium,
+                  fontSize = TEXT_FONT_SIZE.sp,
+                  color = MaterialTheme.colorScheme.onPrimary)
+              Counter(
+                  count = quantity,
+                  onCounterChange = { quantity = it },
+                  modifier = Modifier.background(Color.Transparent))
```

Comment:
Here as well, I find the button to be a bit too large. However, this is subjective, so gathering some user feedback might be helpful. (I did a quick test with a random user who also mentioned the same thing, but this should be discussed with the Product Owner for a final decision.)

Reply:
Thanks for the feedback! I’ll adjust the button size in a follow-up PR next week. It’s good to know that a quick user test echoed this observation. I’ll also discuss it with the Product Owner to align on the final decision.

Reply:
Ok !